### PR TITLE
📊 OECD PISA dataset on educational attainment

### PR DESCRIPTION
**Pull request by veronikasamborska1994**
hey @spoonerf! no rush to review this as I am going to be away for the next couple of days anyway :) One weird issue/bug I have is that my dataset is being automatically saved as "pisa_math_boys_girls" instead of "pisa" in the meadow step. do you by any chance know why/how this is happening? thank you :)

### CODE REVIEW

File: etl/steps/data/garden/oecd/2023-12-06/pisa.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            "pisa_science_se_girls",
+            "pisa_science_se_boys",
+            "pisa_math_se_girls",
+            "pisa_math_se_boys",
+            "pisa_math_all_se",
+            "pisa_science_all_se",
+            "pisa_reading_all_se",
+        ]
+    )
+    # Add metadata.
```
Comment (spoonerf) [inline etl/steps/data/garden/oecd/2023-12-06/pisa.py:47]: I wouldn't change what you've done here but I would recommend trying out the Jinja templates for metadata. 

It's essentially the same as what you've done but I find it somehow a little easier to work with. The main change is that you need wider tables, so a column for each disaggregation you would have different metadata for.

File: etl/steps/data/garden/oecd/2023-12-06/pisa.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# NOTE: To learn more about the fields, hover over their names.
```
Comment (spoonerf) [inline etl/steps/data/garden/oecd/2023-12-06/pisa.yml:1]: I think this filename is missing the `metadata` here, so the tag is not being propagated.

File: snapshots/oecd/2023-12-06/pisa_math_all.xls.dvc [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    producer: OECD
+    citation_full: OECD Programme for International Student Assessment (PISA) (2023),
+      https://www.oecd.org/pisa/data/, accessed on 6th December 2023.
+    # Files
+    url_main: https://www.oecd.org/pisa/data/
+    date_accessed: 2023-12-06
+
+    # License
+    license:
+      name: ©producer year
```
Comment (spoonerf) [inline snapshots/oecd/2023-12-06/pisa_math_all.xls.dvc:27]: Is this the name of the license?

File: etl/steps/data/meadow/oecd/2023-12-06/pisa.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        tbs_both_sexes.append(tb)
+
+    # Merge all dataframes for both sexes into one
+    tbs_both_sexes = reduce(
+        lambda left, right: pr.merge(left, right, on=["country", "year"], how="outer"), tbs_both_sexes
+    )
+
+    # Load and process data for boys and girls separately
+    tbs_split = []
+    for snap in snaps_split_by_sex:
```
Comment (spoonerf) [inline etl/steps/data/meadow/oecd/2023-12-06/pisa.py:64]: Nice handling of not very nicely formatted data!

**Review by spoonerf — APPROVED** [review]
Looks great! 

I'm not sure why the meadow step was automatically changing it's name - it didn't happen to me locally. Maybe something to do with wizard, could be auto-filling something behind the scenes?

File: etl/steps/data/garden/oecd/2023-12-06/pisa.yml (hunk 2 above)
Comment (veronikasamborska1994) [reply etl/steps/data/garden/oecd/2023-12-06/pisa.yml:1]: thanks a lot for spotting this!!

File: snapshots/oecd/2023-12-06/pisa_math_all.xls.dvc (hunk 3 above)
Comment (veronikasamborska1994) [reply snapshots/oecd/2023-12-06/pisa_math_all.xls.dvc:27]: thanks! also, a bug :)

**Pull request merged**